### ISSUE TITLE
Unexpected panic using xml-rs

### ISSUE THREAD

**Issue by rosscanning** [rust-lang/rust#32458]
## Summary

This code:
```rust
extern crate xml;
//
// Dependencies list from Cargo.toml:
// [dependencies]
// xml-rs = "0.3"
//
use xml::reader::{Events, EventReader, XmlEvent};
fn get_operation_name(e: &XmlEvent::StartElement) -> String {
}
```
causes rustc to exit with "unexpected panic".

**Compiler version** is rustc 1.7.0 (a5d1e7a59 2016-02-29).

**Platform** is Windows 8.1, 64-bit.

Possibly related to  #32335.

## Backtrace:
stack backtrace:
   0:     0x7ff8490c56e5 - sys::backtrace::Cleanup.Drop::drop::hb0110d471b4dd2c7Wfu
   1:     0x7ff8490cedc2 - panicking::take_handler::he1d461639b318781oQy
   2:     0x7ff849094358 - sys_common::unwind::begin_unwind_inner::h485629376f8607efSat
   3:     0x7ff849094beb - sys_common::unwind::begin_unwind_fmt::ha23b0b11dd517285Y9s
   4:     0x7ff8490c2592 - rust_begin_unwind
   5:     0x7ff8490d8ee8 - panicking::panic_fmt::h3dc40a8b303af4f9zYL
   6:     0x7ff8490d97c0 - panicking::panic::hf4e33ac4547ee3e16WL
   7:     0x7ff843a9175e - check::method::exists::ha86d9cfde4d0bb4eG6i
   8:     0x7ff843a40ddb - check::FnCtxt<'a, 'tcx>::register_predicate::hf43a0d7e2167b808Adq
   9:     0x7ff843a4006e - check::FnCtxt<'a, 'tcx>::register_predicate::hf43a0d7e2167b808Adq
  10:     0x7ff843b02e67 - rscope::ShiftedRscope<'r>.RegionScope::anon_regions::h0593918629aea6daJvu
  11:     0x7ff843b01f70 - rscope::ShiftedRscope<'r>.RegionScope::anon_regions::h0593918629aea6daJvu
  12:     0x7ff843b20220 - collect::hir..Generics.GetTypeParameterBounds<'tcx>::get_type_parameter_bounds::hf41e00c45292d87dnGx
  13:     0x7ff843b18d25 - collect::hir..Generics.GetTypeParameterBounds<'tcx>::get_type_parameter_bounds::hf41e00c45292d87dnGx
  14:     0x7ff843b04d9f - collect::AstConvRequest...std..clone..Clone::clone::h2b8e47faea33d718xix
  15:     0x7ff843b04bcf - collect::CollectItemTypesVisitor<'a, 'tcx>.intravisit..Visitor<'v>::visit_item::h732b92b6b3ca6ed7Pjx
  16:     0x7ff843a083dc - collect::collect_item_types::h9758abe543620d58dcx
  17:     0x7ff843a0453d - check_crate::h5e5e625dcc1f464drNC
  18:     0x7ff858a9b86e - driver::make_map::h872d102a37abf3cdUMa
  19:     0x7ff858a998e2 - driver::make_map::h872d102a37abf3cdUMa
  20:     0x7ff858a94f53 - driver::make_map::h872d102a37abf3cdUMa
  21:     0x7ff858a64a67 - driver::compile_input::he007aa27013a394bnca
  22:     0x7ff858a55e14 - run_compiler::hc51aca95092a62e87wc
  23:     0x7ff858a533be - run::hdbc586a8c89951e1Nwc
  24:     0x7ff8490bb6bf - sys_common::unwind::try::inner_try::hf61c3446f7caab3aq7s
  25:     0x7ff858a5389c - run::hdbc586a8c89951e1Nwc
  26:     0x7ff8490ccfcd - sys::process::Command::cwd::h01e23a1d066553faVCx
  27:     0x7ff866c013d1 - BaseThreadInitThunk

**Comment by rosscanning** [comment]
**Note**: Does **not** crash with the nightly build dated 2016-03-24, so it may have been fixed already. I'll check again when 1.8 stable is released.

**Comment by rosscanning** [comment]
This is fixed in 1.8 stable.
